NeosModLoader/NeosVersionReset.cs

```diff
@@ -12,65 +15,162 @@ namespace NeosModLoader
 {
 	internal class NeosVersionReset
 	{
+		// used when AdvertiseVersion == true
+		private const string NEOS_MOD_LOADER = "NeosModLoader.dll";
+
 		internal static void Initialize()
 		{
 			ModLoaderConfiguration config = ModLoaderConfiguration.Get();
 			Engine engine = Engine.Current;
 
+			// get the version string before we mess with it
+			string originalVersionString = engine.VersionString;
+
 			List<string> extraAssemblies = Engine.ExtraAssemblies;
 			string assemblyFilename = Path.GetFileName(Assembly.GetExecutingAssembly().Location);
-			bool nmlPresent = extraAssemblies.Contains(assemblyFilename);
+			bool nmlPresent = extraAssemblies.Remove(assemblyFilename);
 
 			if (!nmlPresent)
 			{
 				throw new Exception($"Assertion failed: Engine.ExtraAssemblies did not contain \"{assemblyFilename}\"");
 			}
 
-			bool otherPluginsPresent = extraAssemblies.Count > 1;
-			bool shouldSpoofCompatibility = !otherPluginsPresent || config.Unsafe;
-			bool shouldSpoofVersion = !config.AdvertiseVersion && shouldSpoofCompatibility;
+			// get all PostX'd assemblies. This is useful, as plugins can't NOT be PostX'd.
```

Comment:
some plugin authors use external tools to pre postX their plugins

Reply:
It's either PostX'd or not, and it doesn't matter if it's been done a month ago or seconds ago by neos.